pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -82,6 +82,15 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 			// Arbitrarily grab the very first pod, as they all should be crashing
 			pod := pods.Items[0]
 
+			// Update the revision status if pod cannot be scheduled(possibly resource constraints)
+			// If pod cannot be scheduled then we expect the container status to be empty.
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodScheduled && cond.Status == corev1.ConditionFalse {
+					rev.Status.MarkResourcesUnavailable(cond.Reason, cond.Message)
+					break
+				}
+			}
+
```

Comment:
do we care about overriding this status with the statuses below if they all happen?

Reply:
Do you mean updating revision status if resources are present on nodes?

Reply:
NM, I think it's fine. But what I meant is code below would override the status value. But we don't have idea of priorities, so I guess it doesn't matter here.